src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -177,14 +177,19 @@ public synchronized void open() throws CouchbaseLiteException {
         boolean isOpenSuccess = false;
         try {
             // Open database:
-            storageEngine.open(path);
-            // Try to decrypt or access the database:
-            decrypt(encryptionKey);
+            storageEngine.open(path, encryptionKey);
             isOpenSuccess = true;
         } catch (SQLException e) {
             String message = "Unable to create a storage engine";
             Log.e(TAG, message, e);
-            throw new CouchbaseLiteException(message, e, Status.DB_ERROR);
+            int statusCode;
+            if (e.getCode() == SQLException.SQLITE_ENCRYPTION_UNAUTHORIZED)
```

Comment:
SQLException.SQLITE_ENCRYPTION_UNAUTHORIZED and Status.UNAUTHORIZED are same value.
SQLException. SQLITE_ENCRYPTION_NOTAVAILABLE and Status. NOT_IMPLEMENTED are same value.

The code could be simplified.